### DISCUSSION TITLE
405: Only the HEAD or GET methods are allowed.

### DISCUSSION THREAD

**Discussion by kimmy-wang** [ether/etherpad-lite#5125] [Q&A] [answered]
Cannot send `post` request to visit `http://localhost:9001/api/mention/user/create`, return `405: Only the HEAD or GET methods are allowed`.
How to set up to send post request？

![Lark20210722-115943](https://user-images.githubusercontent.com/19725091/126588316-98cc6571-a2e6-4d34-8654-02638eac7915.png)

**Comment by webzwo0i** [comment, marked as answer]
Your route ends up being served by yajsml which only allows GET/HEAD.

Can you try another path (not /api)?

**Reply by kimmy-wang** [reply]
I modified my routing and removed /api, but it still doesn’t work

**Reply by webzwo0i** [reply]
```
diff --git a/src/node/hooks/express/apicalls.js b/src/node/hooks/express/apicalls.js
index b72ed11e..ab365c23 100644
--- a/src/node/hooks/express/apicalls.js
+++ b/src/node/hooks/express/apicalls.js
@@ -13,6 +13,12 @@ exports.expressCreateServer = (hookName, args, cb) => {
       res.end('OK');
     });
   });
+  args.app.post('/api/shouldnotfail', (req, res) => {
+    console.log('test');
+    new formidable.IncomingForm().parse(req, (err, fields, files) => {
+      res.end('OK');
+    });
+  });
```

This works for me. I noticed you used a new file "src/node/hooks/express/api.js" to register the routes. Did you add the file to src/ep.json so the route is initialized? And if yes, did you put it above or below the line that references `"expressCreateServer": "ep_etherpad-lite/node/hooks/express/static"` ?

You need to put it *below* that line, because in static we register the yajsml middleware, and the error indicates that the middleware is handling the request before your route does.

**Reply by kimmy-wang** [reply]
It works. Thanks
